examples/base/static_vars_walkthrough.cc

```diff
@@ -0,0 +1,326 @@
+            << "Bob.individual   = " << Bob.individual << "\n\n";
+
+  Alice.same = 200; // For both Alice & Bob the variable `same` is 200.
+  Bob.same = 6; // Overrides Alice's setting, for both now `same` is 6!
+  std::cout << "Alice.same = " << Alice.same << "\n"
+            << "Bob.same   = " << Bob.same << "\n\n";
+}
+
+/**
+ * (2) Macros
+ * Macros are preprocessor instructions, that get evaluated during compile time.
+ * It can be used for declaring variables, replace parts of your code or
+ * for writing functions (see (4)).
+ */
+# define X 1
+# define Y "Hello there!"
+# define Z                        \
+  struct reindeer {               \
+    std::string name = "Rudolph"; \
+    unsigned int mph = 200;       \
+  };
+
+Z // define the class reindeer
+
+// Or we can declare functions:
+// as one-liner
+# define F(a, b) ((a + b) % 5)
+// or more complex
+# define G(a, b) int myadd() { int c = (a + b) % 5; \
+                               return c - 1;        \
+                             }
+// Define myadd with a = X = 1 and b = 27.
+// Note, that (a) we omit the semicolon, the compiler adds it automatically
+// and (b) that this cannot be inside a function, since G is a "placeholder"
+// for the full function definition of myadd.
+G(X, 27)
+
+// Some syntactic sugar..
+// -- `#msg` stringifies: msg -> "msg"
+# define H(msg) void mymsg() { std::cout << #msg << "\n"; }
+H(barbershop)
+// -- `a##b` concatenates a and b
+# define I(a,b) int my##a() { return b; }
+I(price,4000) // defines a function myprice that returns 4000
+
+void part2() {
+  std::cout << "\n-- Part 2: Macros --------------------------------------\n\n";
+  std::cout << Y << "\n";
+  std::cout << "X + 2               = " << X + 2 << "\n";
+  std::cout << "F(X, 27)            = " << F(X, 27) << "\n";
+  std::cout << "myadd() [a=X, b=27] = " << myadd() << "\n\n";
+
+  reindeer r;
+  std::cout << r.name << " goes " << r.mph << " mph!\n\n";
+
+  std::cout << "I went to a [mymsg()] ";
+  mymsg();
+  std::cout << "and that cost me [myprice()]" << myprice() << " CHF.\n";
+}
+
+/**
+ * (3) Track-Class
+ * Basically a linked list with a stored value per element.
+ * The root element is the one that's added the latest.
+ */
+template <class T>
+class Track {
+  public:
+   Track *next_;         // next variable in list
+   Track *prev_;         // previous variable in list
+   std::string name_;    // name of variable
+   std::string comment_; // optional documentation
+   T &ref_;              // stored value
+
+   /**
+    * Constructor: Places a new item of the global info list in the list
+    * Usually called via a macro (DECLARE, COUNTER).
+    * The second version of the constructor also permits to add a comment to the
+    * information item.
+    */
+   Track(std::string name, T &ref, Track<T> *&root,
+         std::string comment = std::string({}));
+   Track() = delete;      // tells compiler to not create a default constructor
```

Comment:
Sidemark: Not needed since you've declared another constructor ;)

Reply:
Thanks! I'm learning with your comments at least as much as the students should have via this demo :smile: (file deleted though)